signer/kms/aws/client.go

```diff
@@ -0,0 +1,318 @@
+	}
+	return nil, lerr
+}
+
+// At the moment this function lies unused, but it is here for future if necessary
+func (a *awsClient) verify(ctx context.Context, sig, message io.Reader) error {
+	cmk, err := a.getCMK(ctx)
+	if err != nil {
+		return err
+	}
+
+	verifier, err := cmk.Verifier()
+	if err != nil {
+		return err
+	}
+
+	s, err := io.ReadAll(sig)
+	if err != nil {
+		return err
+	}
+
+	return verifier.Verify(message, s)
+}
+
+func (a *awsClient) verifyRemotely(ctx context.Context, sig, digest []byte) error {
+	cmk, err := a.getCMK(ctx)
+	if err != nil {
+		return err
+	}
+
+	alg := cmk.KeyMetadata.SigningAlgorithms[0]
+	messageType := types.MessageTypeDigest
+	if _, err := a.client.Verify(ctx, &akms.VerifyInput{
+		KeyId:            &a.keyID,
+		Message:          digest,
+		MessageType:      messageType,
+		Signature:        sig,
+		SigningAlgorithm: alg,
+	}); err != nil {
+		return fmt.Errorf("unable to verify signature: %w", err)
+	}
+
+	return nil
+}
+
+func (a *awsClient) sign(ctx context.Context, digest []byte, _ crypto.Hash) ([]byte, error) {
+	cmk, err := a.getCMK(ctx)
+	if err != nil {
+		return nil, err
+	}
+	alg := cmk.KeyMetadata.SigningAlgorithms[0]
+
+	messageType := types.MessageTypeDigest
+	out, err := a.client.Sign(ctx, &akms.SignInput{
+		KeyId:            &a.keyID,
+		Message:          digest,
+		MessageType:      messageType,
+		SigningAlgorithm: alg,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("signing with kms: %w", err)
+	}
+	return out.Signature, nil
+}
+
+func (a *awsClient) fetchPublicKey(ctx context.Context) (crypto.PublicKey, error) {
+	out, err := a.client.GetPublicKey(ctx, &akms.GetPublicKeyInput{
+		KeyId: &a.keyID,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("getting public key: %w", err)
+	}
+	key, err := x509.ParsePKIXPublicKey(out.PublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("parsing public key: %w", err)
+	}
+	return key, nil
+}
+
+func (a *awsClient) fetchKeyMetadata(ctx context.Context) (*types.KeyMetadata, error) {
+	out, err := a.client.DescribeKey(ctx, &akms.DescribeKeyInput{
+		KeyId: &a.keyID,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("getting key metadata: %w", err)
+	}
+	return out.KeyMetadata, nil
+}
+
+type cmk struct {
+	KeyMetadata *types.KeyMetadata
+	PublicKey   crypto.PublicKey
+}
+
+func (c *cmk) HashFunc() crypto.Hash {
+	switch c.KeyMetadata.SigningAlgorithms[0] {
+	case types.SigningAlgorithmSpecRsassaPssSha256, types.SigningAlgorithmSpecRsassaPkcs1V15Sha256, types.SigningAlgorithmSpecEcdsaSha256:
+		return crypto.SHA256
+	case types.SigningAlgorithmSpecRsassaPssSha384, types.SigningAlgorithmSpecRsassaPkcs1V15Sha384, types.SigningAlgorithmSpecEcdsaSha384:
+		return crypto.SHA384
+	case types.SigningAlgorithmSpecRsassaPssSha512, types.SigningAlgorithmSpecRsassaPkcs1V15Sha512, types.SigningAlgorithmSpecEcdsaSha512:
+		return crypto.SHA512
+	default:
+		return 0
+	}
+}
+
+func (c *cmk) Verifier() (cryptoutil.Verifier, error) {
+	return c.Verifier()
```

Comment:
Looks like this entire function was taken out. All that's left is a recursive call. 😊